amr-wind/overset/OversetOps.cpp

```diff
@@ -0,0 +1,169 @@
+#include "amr-wind/overset/OversetOps.H"
+#include "amr-wind/core/field_ops.H"
+#include "AMReX_ParmParse.H"
+#include "AMReX_MultiFabUtil.H"
+#include "amr-wind/core/MLMGOptions.H"
+#include "amr-wind/projection/nodal_projection_ops.H"
+#include <hydro_NodalProjector.H>
+
+namespace amr_wind {
+
+void OversetOps::initialize(CFDSim& sim) { m_sim_ptr = &sim; }
+
+void OversetOps::pre_advance_actions()
+{
+    // Update pressure gradient using updated overset pressure field
+    update_gradp();
+}
+
+void OversetOps::update_gradp()
+{
+    BL_PROFILE("amr-wind::OversetOps::update_gradp");
+
+    auto density = (*m_sim_ptr).repo().get_field("density").vec_const_ptrs();
+    auto scaling_factor = (*m_sim_ptr).time().deltaT();
+
+    // Pressure and sigma are necessary to calculate the pressure gradient
+
+    const bool is_anelastic = (*m_sim_ptr).is_anelastic();
+    const bool variable_density =
+        (!(*m_sim_ptr).pde_manager().constant_density() ||
+         (*m_sim_ptr).physics_manager().contains("MultiPhase"));
+
+    bool mesh_mapping = (*m_sim_ptr).has_mesh_mapping();
+
+    auto& grad_p = (*m_sim_ptr).repo().get_field("gp");
+    auto& pressure = (*m_sim_ptr).repo().get_field("p");
+    auto& velocity = (*m_sim_ptr).repo().get_field("velocity");
+    amr_wind::Field const* mesh_fac =
+        mesh_mapping ? &((*m_sim_ptr)
+                             .repo()
+                             .get_mesh_mapping_field(amr_wind::FieldLoc::CELL))
+                     : nullptr;
+    amr_wind::Field const* mesh_detJ =
+        mesh_mapping ? &((*m_sim_ptr)
+                             .repo()
+                             .get_mesh_mapping_detJ(amr_wind::FieldLoc::CELL))
+                     : nullptr;
+    const auto* ref_density =
+        is_anelastic ? &((*m_sim_ptr).repo().get_field("reference_density"))
+                     : nullptr;
+
+    // ASA does not think we actually need to define sigma here. It
```

Comment:
this PR would be a good opportunity to check if the sigma calculation is needed and remove it

Reply:
confirmed it is unnecessary and removed it